src/Orleans.Core/Messaging/CachingSiloAddressCodec.cs

```diff
@@ -114,7 +115,7 @@ private static SiloAddress ReadSiloAddressInner<TInput>(ref Reader<TInput> reade
                 ip = new IPAddress(addressBytes);
             }
             var port = (int)reader.ReadVarUInt32();
-            var generation = reader.ReadInt32();
+            var generation = reader.ReadVarInt32();
```

Comment:
I used a fixed-width here originally because that is faster and occupies 4 bytes on the wire, whereas a var-int would not occupy less than 4 bytes and would be slower to encode & decode.

``` C#
void Main()
{
	AllocateNewGeneration().ToString("X").Dump(); // "179C9C1B" when I checked
}

private static readonly long epoch = new DateTime(2010, 1, 1).Ticks;

public static int AllocateNewGeneration()
{
    long elapsed = (DateTime.UtcNow.Ticks - epoch) / TimeSpan.TicksPerSecond;
    return unchecked((int)elapsed); // Unchecked to truncate any bits beyond the lower 32
}
```